.github/workflows/lint.yml

```diff
@@ -0,0 +1,17 @@
+name: Lint
+
+on:
+  pull_request:
+    branches: [main]
```

Comment:
Should we also add paths-ignore since lint is for code not docs?

Reply:
Not sure if we need to run lint for .github and .tekton since these are excluded in https://github.com/kserve/modelmesh-serving/blob/c77a44f3ed47be7b2d340c0f4928537abb7b142e/.pre-commit-config.yaml#L23